src/streamlink_cli/utils/path.py

```diff
@@ -35,6 +35,28 @@ def replace_chars(path: str, charmap: Optional[str] = None, replacement: str = R
     return pattern.sub(replacement, path)
 
 
+# This method does not take care of unicode modifier characters when truncating
+def truncate_path(path: str, length: int = 255, keep_extension: bool = True) -> str:
+    if len(path) <= length:
+        return path
+
+    parts = path.rsplit(".", 1)
+
+    # no file name extension (no dot separator in path or file name extension too long):
+    # truncate the whole thing
+    if not keep_extension or len(parts) == 1 or len(parts[1]) > 10:
+        encoded = path.encode("utf-8")
+        truncated = encoded[:length]
+        decoded = truncated.decode("utf-8", errors="ignore")
+        return decoded
```

Comment:
Btw, the `or len(parts[1]) > 10` check is added here so that only file name extensions with fewer than 11 characters will be kept. We must avoid unnecessary file name truncation in cases where `--output="{title}"` (no user-set file name extension) and when the `title` metadata includes a dot somewhere in its value. Without the check, the whole text after the last dot would be treated as a file name extension and always be kept, unnecessarily reducing the available space before of the last dot.